Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1908,17 +2030,23 @@ var event = function(
         typeof recEvt === 'object',
         'Bad event of type ' + typeof recEvt + ' for key ' + key
       );
-      for (var key in recMapping) {
-        traverse(recMapping[key], recEvt[key], key);
+      for (const mappingKey in recMapping) {
+        traverse(recMapping[mappingKey], recEvt[mappingKey], mappingKey);
       }
     };
-    argMapping.forEach((mapping, idx) => {
-      traverse(mapping, args[idx], 'arg' + idx);
-    });
-    if (config && config.listener) {
-      config.listener.apply(null, args);
-    }
-  };
+  }
+}
+
+var event = function(
+  argMapping: Array<?Mapping>,
+  config?: ?EventConfig,
+): any {
```

Comment:
Would be good if we could return function from this method. Otherwise it may result in react warnings as `onScroll` is defined to only accept functions (not sure if you're getting them or not).

I see two options to solve this:
1) when event is native, we return a method that will throw an exception when called (handler should never be called as we expect event to be handled in native). Then we can bind some extra arguments to the function object so that we can recognise if the handler is native or not in `AnimatedProps.__getValue` (its a little bit "hacky")

2) Somehow make `Animated.Scroll` wrapper override proptypes so that it allows both function and AnimatedEvent object